lib/OpenQA/Task/Needle/Save.pm

```diff
@@ -101,10 +102,13 @@ sub _save_needle {
 
     # ensure needle dir is up-to-date
     my $git = OpenQA::Git->new({app => $app, dir => $needledir, user => $user});
+    # avoid retrying if cleanup via Git is not configured (then an admin might need to
+    # manually cleanup the repo first)
+    $signal_guard->abort(1) if !$git->enabled || $git->config->{do_cleanup} eq 'no';
     if ($git->enabled) {
         my $error = $git->set_to_latest_master;
         if ($error) {
-            $app->log->error($error);
+            $app->log->error("Dirty repo detected, a manual cleanup is required. Error: $error");
```

Comment:
The previous call might have failed due to a different reason. I suggest you simply take this one line back.

Reply:
AC3 is more about the error message that the user will see in the web browser.
And like @Martchus said, we don't know here if the error is actually about a dirty git status.

The error message the user would see is the error passed when calling `$minion_job->fail(...)`.
We discussed yesterday that setting `abort` to true will make the minion job finish successfully, so right now AC3 cannot be fulfilled. I guess the signalhandler class would have to support that.